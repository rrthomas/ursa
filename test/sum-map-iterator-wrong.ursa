let sum = fn(m) {
  let tot = 0
  let it = m.iterator()
  loop {
    let l = it()
    let k = l[0] and let v = l[1]
    if l == null { return tot }
    print(k)
    tot := tot + v
  }
}
sum({"a": 10, "b": 30, "c": 50, "d": 5, "e": 5})
