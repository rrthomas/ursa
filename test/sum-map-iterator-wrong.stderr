Error: Line 6, col 13:
  5 |     let l = it()
> 6 |     let k = l.get(0) and let v = l.get(1)
                  ^~~~~
  7 |     if l == null { return tot }

Invalid property

Traceback (most recent call last)
  line 12
    sum({"a": 10, "b": 30, "c": 50, "d": 5, "e": 5}), at top level